### PR TITLE
Fix non-zero exit code on script error

### PR DESCRIPTION
**Pull request by moritzraho**
## Description
Return a non zero error code on error in scripts

## Motivation and Context
Most scripts do not return a non zero error code, which leads to Travis false positives, leads to cmds combining scripts with `&&` to execute the second script even if the first failed and so on ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: scripts/deploy.actions.js [hunk 1]
```diff
@@ -50,7 +50,6 @@ function deployActionsSync () {
   // for now this is a tmp hack so that ~/.wskprops does not interfer with WHISK_* properties defined in .env
   const fakeWskProps = '.fake-wskprops'
   fs.writeFileSync(fakeWskProps, '')
-
   process.env['WSK_CONFIG_FILE'] = fakeWskProps
   // aio reads env WHISK_* properties
   const aio = spawn.sync(
```
Comment (rajarju) [inline scripts/deploy.actions.js:56]: @moritzraho Doesn't spawn.sync throw an error if the command exists with non zero error ?
Do we have to explicitly check and throw again asin line 66 and 67

File: scripts/generate.config.js [hunk 2]
```diff
@@ -23,4 +23,9 @@ function generateConfig () {
   )
 }
 
-generateConfig()
```
Comment (rajarju) [inline scripts/generate.config.js:26]: Isn't there too many try catch in the code, which all handle the errors the same way.
Can't they be all handled at a higher level ?

File: scripts/deploy.actions.js (hunk 1 above)
Comment (moritzraho) [reply scripts/deploy.actions.js:56]: @rajarju yes pretty sure we have to. To be sure I tried to inject an error in the command and remove 66 and 67 and the error wasn't catch.

File: scripts/generate.config.js (hunk 2 above)
Comment (moritzraho) [reply scripts/generate.config.js:26]: Yep thanks I agree I will submit a commit

**Review by rajarju — APPROVED** [review]
LGTM

**Pull request merged**